### DISCUSSION TITLE
Make sloped terrain behave like flat terrain

### DISCUSSION THREAD

**Discussion by JimmyThrush** [Techokami/SonicWorldsNext#110] [Q&A] [answered]
Certain moving platforms in the Genesis games, like the moving sloped platforms in Marble Zone, or the rising floor in Hill Top Zone, are sloped in appearance and do change Sonic's elevation, but Sonic doesn't have any of his other typical interactions with slopes; IE, running or rolling downwards doesn't speed him up, he's not slowed down by running upwards, he doesn't gain any elevation from jumping off the upwards slope, etc. A demonstration of this in webm form is attached to the post (it's from Sonic 1 Forever, but it behaves the same in the original Genesis rom).

Can this behavior be recreated in Sonic Worlds Next? I suppose you could add a collision layer and make it so that Sonic's slope interactions are skipped when Sonic is colliding with that layer and that could be an okay way of doing it. Any other ideas for solutions?

[test.webm](https://github.com/Techokami/SonicWorldsNext/assets/123826402/c4478797-9616-4d92-bf44-349bfaa88990)

**Comment by DimensionWarped** [comment, marked as answer]
According to @Renhoex you can accomplish this with the 'object layer'. Look at the teleporter's orb object for an example.

![image](https://github.com/Techokami/SonicWorldsNext/assets/7477617/d394c0eb-b415-438b-ae8f-8084a64525e5)
